packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -903,21 +902,30 @@ class ServiceExtensionManager extends Disposer {
   }
 
   Future<void> _onMainIsolateChanged() async {
-    if (_mainIsolate.value == null) {
+    if (_isolateManager.mainIsolate.value == null) {
       _mainIsolateClosed();
       return;
     }
     _checkForFirstFrameStarted = false;
 
-    final isolateRef = _mainIsolate.value;
-    final Isolate isolate = await _service.getIsolate(isolateRef.id);
-    if (isolateRef != _mainIsolate.value) {
+    final isolateRef = _isolateManager.mainIsolate.value;
+    final Isolate isolate = await _isolateManager.getIsolateCached(isolateRef);
+
+    await _registerIsolate(isolate, isolateRef);
+  }
+
+  Future<void> _registerIsolate(Isolate isolate,
+      [IsolateRef previousIsolateRef]) async {
```

Comment:
in general avoid optional positional parameters. they just lead to bugs when they are sometimes used and sometimes not used. In general prefer optional named parameters as they provide more clarity about the intent of the parameters for cases where optional parameters are needed. not sure if this is in the style guide but is implicitly the case for flutter.